packages/orm/src/relations/hasPersistedAsyncRelation.ts

```diff
@@ -0,0 +1,342 @@
+
+import { AbstractRelationImpl } from "./AbstractRelationImpl";
+import { RelationT, RelationU } from "./Relation";
+import { Reference, ReferenceN } from "./Reference";
+import { Entity } from "../Entity";
+import { Reacted, ReactiveHint } from "../reactiveHints";
+import { Const, IdOf, currentlyInstantiatingEntity } from "../EntityManager";
+import { EntityMetadata, ManyToOneField, OneToManyCollection, OneToManyLargeCollection, OneToOneReferenceImpl, deTagId, deTagIds, ensureNotDeleted, ensureTagged, getMetadata, isEntity, isLoaded, maybeResolveReferenceToId, sameEntity, setField } from "..";
+
+const I = Symbol();
+
+export interface PersistedAsyncRelation<T extends Entity, U extends Entity, N extends never | undefined> extends Reference<T, U, N> {
+  isLoaded: boolean;
+  isSet: boolean;
+
+  /**
+   * Returns the as-of-last-flush previously-calculated entity.
+   *
+   * This is useful if you have to purposefully avoid using the lambda to calc the latest entity,
+   * i.e. if you're in a test and want to watch a calculated entity change from some dummy entity
+   * to the new derived entity.
+   * */
+  fieldValue: U | N;
```

Comment:
Thought: Maybe we can skip this--this does actually match PersistedAsyncProperty, so lets keep for now.